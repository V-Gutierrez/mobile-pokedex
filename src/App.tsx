import { Header } from './components'
import { PokemonList } from './views'
import React from 'react'
import {
  SafeAreaView,
  StyleSheet,
} from 'react-native'


const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
})

const App = () => {
  return (
    <SafeAreaView style={styles.container}>
      <Header />
      <PokemonList />
    </SafeAreaView>
  )
}


export default App
