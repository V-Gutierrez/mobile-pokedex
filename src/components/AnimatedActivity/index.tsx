import React from 'react';
import { ActivityIndicator, Animated, Easing } from 'react-native';


export const AnimatedActivity: React.FC = () => {
  const rotation = new Animated.Value(0)

  const rotate = rotation.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg']
  })

  rotation.setValue(0)
  Animated.timing(rotation, {
    toValue: 1,
    duration: 1080,
    easing: Easing.linear,
    useNativeDriver: true
  }).start()

  return (
    <Animated.View style={{ justifyContent: 'center', alignItems: 'center', transform: [{ rotate }] }}>
      <ActivityIndicator size='large' color='gray' />
    </Animated.View>
  )
}
