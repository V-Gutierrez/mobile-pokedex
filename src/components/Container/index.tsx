import React from 'react';
import {SafeAreaView, StatusBar, StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#3D7DCA',
  },
});

interface Props {
  children: React.ReactNode;
}

export const Container: React.FC<Props> = ({children}) => {
  return (
    <SafeAreaView style={styles.container}>
      <StatusBar backgroundColor="#003A70" />
      {children}
    </SafeAreaView>
  );
};
