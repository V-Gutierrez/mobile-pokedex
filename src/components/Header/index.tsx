import React from 'react';
import {StyleSheet, Text, View, Image} from 'react-native';

const styles = StyleSheet.create({
  container: {
    height: '25%',
    width: '100%',
    paddingVertical: 20,
    paddingHorizontal: 10,
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 25,
    backgroundColor: '#003A70',
  },
  headerTitle: {
    fontSize: 24,
    justifyContent: 'center',
    marginLeft: 20,
    fontWeight: 'bold',
    color: '#FFCB05',
    fontFamily: 'Pokemon',
  },
  pikachu: {
    zIndex: -1,
    top: 35,
  },
  searchIcon: {
    alignItems: 'center',
    borderRadius: 10,
    marginLeft: 15,
  },
  titleAndSearchWrapper: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
  },
});

export const Header: React.FC = () => {
  return (
    <View style={styles.container}>
      <View style={styles.titleAndSearchWrapper}>
        <Text style={styles.headerTitle} testID="header-title">
          Mobile Pokedex
        </Text>
      </View>
      <View>
        <Image
          source={require('../../assets/images/025.png')}
          style={styles.pikachu}
        />
      </View>
    </View>
  );
};
