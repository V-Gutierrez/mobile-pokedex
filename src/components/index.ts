export * from './Header'
export * from './PokemonCard'
export * from './AnimatedActivity'
export * from './PokemonModal'