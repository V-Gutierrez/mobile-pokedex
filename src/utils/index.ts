export * from './parseAssetUrl';
export * from './capitalizeStringFirstIndex';
