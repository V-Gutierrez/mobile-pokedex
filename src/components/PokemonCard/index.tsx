import React from 'react'
import { StyleSheet, View, Text, Image, TouchableOpacity } from 'react-native'
import { PokemonInitialData } from 'types'
import { parseAssetUrl } from 'utils'

const styles = StyleSheet.create({
  card: {
    backgroundColor: '#CCCC',
    height: 150,
    width: 170,
    padding: 20,
    marginVertical: 15,
    marginHorizontal: 12,
    borderRadius: 10,
  },
  title: {
    fontSize: 22,
    color: '#FFF',
    fontWeight: 'bold',
    zIndex: 1
  },
  pokemonImage: {
    width: 130,
    height: 130,
  },
  pokemonImageWrapper: {
    backgroundColor: '#FFFFFF',
    borderTopLeftRadius: 100,
    borderBottomLeftRadius: 100,
    borderTopRightRadius: 100,
    width: 110,
    height: 110,
    position: 'absolute',
    top: 50,
    left: 70,
    zIndex: -1,
    alignItems: 'center',
  }
})

export const PokemonCard: React.FC<PokemonInitialData> = ({ name, url, }) => {

  /* TODO: Write link parser here */

  return (
    <TouchableOpacity>
      <View style={styles.card}>
        <Text style={styles.title}>{name}</Text>
        <View style={styles.pokemonImageWrapper}>
          <Image style={styles.pokemonImage} source={{ uri: parseAssetUrl(url) }} />
        </View>
      </View>
    </TouchableOpacity>
  )
}
