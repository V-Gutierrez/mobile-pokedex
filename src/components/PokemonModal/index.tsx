
import useAxios from 'axios-hooks'
import { AnimatedActivity } from '../../components'
import React from 'react'
import { Animated, Easing, Image, ImageBackground, Modal, ModalProps, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { PokemonData } from '../../types'
import { parseAssetUrl, capitalizeStringFirstIndex } from '../../utils'

const breathing = new Animated.Value(0)

Animated.loop(
  Animated.sequence([
    Animated.timing(breathing, {
      toValue: 1.28,
      duration: 1000,
      easing: Easing.linear,
      useNativeDriver: true
    }),
    Animated.timing(breathing, {
      toValue: 1.3,
      duration: 1000,
      easing: Easing.linear,
      useNativeDriver: true
    })
  ])
).start()

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  container: {
    backgroundColor: '#3D7DCA',
    height: '100%',
  },
  pokemonImage: {
    width: 200,
    height: 200,
  },
  pokemonWrapper: {
    width: '100%',
    backgroundColor: '#003A70',
    height: 400,
    borderBottomRightRadius: 100
  },
  pokemonName: {
    fontSize: 36,
    fontWeight: 'bold',
    color: '#FFCB05',
    marginLeft: 15,
    marginTop: 30,
  },
  pokemonStats: {
    height: 25,
  },
  stats: {
    color: '#fff',
    fontWeight: 'bold',
  },
  button: {
    paddingHorizontal: 15,
    paddingVertical: 5,
    marginTop: 15,
    backgroundColor: '#fff',
    borderRadius: 15
  }
})



interface Props extends ModalProps {
  pokemonUrl: string
}


export const PokemonModal: React.FC<Props> = (props) => {
  const { pokemonUrl, onRequestClose } = props
  const [{ data, loading, error }, refetch] = useAxios<PokemonData>(pokemonUrl)


  return (
    <Modal
      animationType='slide'
      presentationStyle='formSheet'
      {...props}
    >
      <View style={styles.container}>

        <View style={styles.pokemonWrapper}>
          <Text style={styles.pokemonName}>{
            loading ? <AnimatedActivity /> : capitalizeStringFirstIndex(data?.name!)}
          </Text>

          <View style={[styles.centeredView]}>

            <Animated.View style={{ transform: [{ scale: breathing }] }} >
              <View style={styles.centeredView}>
                <Image style={styles.pokemonImage} source={{ uri: parseAssetUrl(pokemonUrl) }} />
              </View>
            </Animated.View >

          </View>

        </View>

        <View style={[styles.pokemonStats, styles.centeredView]}>

          <Text style={styles.stats}>Exp. base: {data?.base_experience}</Text>
          <Text style={styles.stats}>Altura: {data?.height}</Text>
          <Text style={styles.stats}>Ordem: {data?.order}</Text>
          <Text style={styles.stats}>Peso: {data?.weight}</Text>

          <TouchableOpacity style={styles.button} onPress={onRequestClose}>
            <Text>Voltar</Text>
          </TouchableOpacity>
        </View>
      </View>
    </Modal >
  )
}
