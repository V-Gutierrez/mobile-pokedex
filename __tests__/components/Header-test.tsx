import 'react-native'
import React from 'react'
import { Header } from '../../src/components'

import { render } from '@testing-library/react-native'


describe('Header component', () => {

  it('should render title correctly', () => {
    const { getByText } = render(
      <Header />
    )
    const headerTitle = getByText('Mobile Pokedex')

    expect(headerTitle).toBeDefined()
  })
})
