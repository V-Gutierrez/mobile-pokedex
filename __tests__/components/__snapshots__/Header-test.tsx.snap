// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#003A70",
      "borderBottomLeftRadius": 25,
      "borderBottomRightRadius": 20,
      "flexDirection": "row",
      "height": "25%",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFCB05",
          "fontFamily": "Pokemon",
          "fontSize": 24,
          "fontWeight": "bold",
          "justifyContent": "center",
          "marginLeft": 20,
        }
      }
    >
      Mobile Pokedex
    </Text>
  </View>
  <View>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/025.png",
        }
      }
      style={
        Object {
          "top": 35,
          "zIndex": -1,
        }
      }
    />
  </View>
</View>
`;
